### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'name')

### ISSUE THREAD

**Issue by micalevisk** [jmcdo29/nestjs-spelunker#31]
### input code

```ts
async function boostrap() {
  await SpelunkerModule.debug(AppModule)
}
```

sorry but I couldn't share the full project and it's a bit larger so I didn't manage to write a minimum repro of this issue. But I'm leaving this here so you (and others) can be aware of this bug.

- `forwardRef()` on few modules & services
- `ModuleRef` on few modules
- no lazy loaded modules
- 16 nestjs modules
- more than 25 nestjs services

### error

when running `nest start` that just uses the debug mode, I got the following output:

![image](https://user-images.githubusercontent.com/13461315/179142994-2a179669-ae5d-4055-8208-340ec8e2bf2a.png)

### env

```
[System Information]
OS Version     : Linux 5.13
NodeJS Version : v16.15.1
NPM Version    : 8.11.0 

[Nest CLI]
Nest CLI Version : 8.2.8 

[Nest Platform Information]
platform-express version : 8.4.7
schematics version       : 8.0.11
swagger version          : 5.2.1
typeorm version          : 8.0.4
testing version          : 8.4.7
common version           : 8.4.7
core version             : 8.4.7
jwt version              : 8.0.1
cli version              : 8.2.8
```

**Comment by jmcdo29** [comment]
Hmm, I can try to look into it, but without having something to help me reproduce the error this might be a bit difficult. Thanks for bringing it up though

**Comment by micalevisk** [comment]
I tried this again and now I got no error lol

**Comment by micalevisk** [comment]
nvm. `npx nest start` raise that error while `npx nest start --webpack` doesn't

**Comment by micalevisk** [comment]
I still didn't manage to find what is wrong but I got another hint while debugging which provider has failed at the following line (ie., `dep === undefined`)

https://github.com/jmcdo29/nestjs-spelunker/blob/158ec4ff7e3269a8fe0cd6b76d075a10cbf9ead7/src/debug.module.ts#L279

Looks like all of the providers are class-based one and they depends on the same provider, which doesn't live on a global module. Since my private project is huge and relies on typeorm, I don't think it would be useful to you. But I could share it with you.

**Comment by micalevisk** [comment]
basically, the following line is returning `undefined` for the second element

https://github.com/jmcdo29/nestjs-spelunker/blob/158ec4ff7e3269a8fe0cd6b76d075a10cbf9ead7/src/debug.module.ts#L274-L279

![image](https://user-images.githubusercontent.com/13461315/222978787-14123af2-f9fd-4839-84e5-42f6d5d480ea.png)

![image](https://user-images.githubusercontent.com/13461315/222978914-398a1f3f-4252-4c81-985b-3d94b91a43a6.png)

which breaks the `dep.name` statement.

**Comment by jmcdo29** [comment]
Awesome. Thank you for the reproduction. That helps out immensely. I'll get this test case added and code fixed up soon :)

**Comment by jmcdo29** [comment]
Fixed in 1.1.5
